include/zephyr/net/lwm2m.h

```diff
@@ -1252,6 +1252,7 @@ enum lwm2m_rd_client_event {
 	LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_TRANSFER_COMPLETE,
 	LWM2M_RD_CLIENT_EVENT_REGISTRATION_FAILURE,
 	LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE,
+	LWM2M_RD_CLIENT_EVENT_REG_TIMEOUT_FAILURE,
```

Comment:
I'd rather be consistent with other regsitration-related events naming, i. e. `LWM2M_RD_CLIENT_EVENT_REGISTRATION_TIMEOUT`